lib/evmone/analysis.cpp

```diff
@@ -25,14 +25,19 @@ struct block_analysis
     /// Close the current block by producing compressed information about the block.
     [[nodiscard]] block_info close() const noexcept
     {
+        using gas_cost_type = decltype(block_info{}.gas_cost);
```

Comment:
Also confusing that "gas_cost" here means two different things, maybe rename this to `final_gas_cost_type` and variable below to `final_gas_cost_max`?

Reply:
I will try to do something about naming here.

Reply:
Refactored. No names any more :)